Documentation/maintenance.md

```diff
@@ -0,0 +1,139 @@
+# Maintenance
+
+## Overview
+
+An etcd cluster needs periodic maintenance to remain reliable.
+Depending on an etcd application's needs, this maintenance can usually
+be automated and performed without down time or significantly degraded performance.
+
+All etcd maintenance manages storage resources consumed by the etcd keyspace.
+Failure to adequately control the keyspace size is guarded by storage space quotas;
+if an etcd member runs low on space, a quota will trigger cluster-wide alarms
+which will put the system into a limited-operation maintenance mode that
+forbids writing new key data.
+To avoid running out of space for writes to the keyspace, the etcd keyspace
+history must be regularly compacted as it ingests writes.
```

Comment:
that's not necessarily true-- if there aren't any writes there's no need to compact